agent/src/plugin_manager/device_plugin_instance_controller.rs

```diff
@@ -0,0 +1,1517 @@
+            .filter_map(|(i, v)| match v {
+                v if v.is_owned_by(&self.node_name) => {
+                    Some((format!("{}-{}", self.instance_name, i), v.to_string()))
+                }
+                _ => None,
+            })
+            .collect();
+        let api = self.kube_client.namespaced(&self.instance_namespace);
+        let patch = Patch::Apply(
+            serde_json::to_value(Object {
+                types: Some(TypeMeta {
+                    api_version: "akri.sh/v0".to_owned(),
+                    kind: "Instance".to_owned(),
+                }),
+                status: None::<NotUsed>,
+                spec: PartialInstanceSlotUsage { device_usage },
+                metadata: ObjectMeta {
+                    name: Some(self.instance_name.to_owned()),
+                    ..Default::default()
+                },
+            })
+            .unwrap(),
+        );
+        api.raw_patch(
+            &self.instance_name,
+            &patch,
+            &PatchParams::apply(&format!("dp-{}", &self.node_name)),
+        )
+        .await
+        .map_err(|e| match e {
+            kube::Error::Api(ae) => match ae.code {
+                409 => {
+                    trace!("Conflict on apply {:?}", ae);
+                    DevicePluginError::SlotInUse
+                }
+                _ => DevicePluginError::Other(ae.into()),
+            },
+            e => DevicePluginError::Other(e.into()),
+        })?;
+        Ok(id)
+    }
+
+    async fn free_slot(&self, id: usize) -> Result<(), DevicePluginError> {
+        let slots_status = self.slots_status.lock().await;
+        slots_status.send_if_modified(|slots| {
+            if id >= slots.len() {
+                // We try to free a slot that doesn't exists, probably already freed
+                false
+            } else {
+                slots[id] = DeviceUsage::Unused;
+                true
+            }
+        });
+        let device_usage = slots_status
+            .borrow()
+            .iter()
+            .enumerate()
+            .filter_map(|(i, v)| match v {
+                v if v.is_owned_by(&self.node_name) => {
+                    Some((format!("{}-{}", self.instance_name, i), v.to_string()))
+                }
+                _ => None,
+            })
+            .collect();
+        let api = self.kube_client.namespaced(&self.instance_namespace);
+        let patch = Patch::Apply(
+            serde_json::to_value(Object {
+                types: Some(TypeMeta {
+                    api_version: "akri.sh/v0".to_owned(),
+                    kind: "Instance".to_owned(),
+                }),
+                status: None::<NotUsed>,
+                spec: PartialInstanceSlotUsage { device_usage },
+                metadata: ObjectMeta {
+                    name: Some(self.instance_name.to_owned()),
+                    ..Default::default()
+                },
+            })
+            .unwrap(),
```

Comment:
We should be avoiding unwraps
```suggestion
            .context("could not create instance patch")?,
```